src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -211,6 +211,20 @@ public GlobalSearchBar(JabRefFrame frame, StateManager stateManager, Preferences
 
         this.stateManager.activeSearchQueryProperty().addListener((obs, oldvalue, newValue) -> newValue.ifPresent(this::updateSearchResultsForQuery));
         this.stateManager.activeDatabaseProperty().addListener((obs, oldValue, newValue) -> stateManager.activeSearchQueryProperty().get().ifPresent(this::updateSearchResultsForQuery));
+        /*
+         * Attach a change listener on the focus property of searchField node.
+         * Every time the listener tracks a change on focus property value,
+         * (i.e. from active (user types query) to inactive / lost (user selects
+         * entry / completes search)), ensures the focus is lost
+         * (i.e. newValue : false), checks whether the search query is
+         * empty or not and finally records it into the search history
+         * through addSearchHistory method call.
+         */
+        searchField.focusedProperty().addListener((obs, oldValue, newValue) -> {
+            if (oldValue && !(newValue || searchField.getText().isEmpty())) {
```

Comment:
Here you can add the detailed comment:

// Focus lost can derived by checking that there is no newValue (or the text is empty)